test/test_hydrostatic_free_surface_immersed_boundaries_apply_surf_bc.jl

```diff
@@ -0,0 +1,79 @@
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBoundary, RasterDepthMask
+using Oceananigans.TurbulenceClosures: VerticallyImplicitTimeDiscretization
+
+@testset "Immersed boundaries with hydrostatic free surface models" begin
+    @info "Testing immersed boundaries with hydrostatic free surface models..."
+
+    for arch in archs
+        Nx = 60
+        Ny = 60
+
+        # A spherical domain
+        underlying_grid = RegularLatitudeLongitudeGrid(size = (Nx, Ny, 1),
+                                               longitude = (-30, 30),
+                                               latitude = (15, 75),
+                                               z = (-4000, 0))
+
+        @inline raster_depth(i, j) = 30 < i < 35 && 42 < j < 48
+
+        grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBoundary(raster_depth, mask_type=RasterDepthMask()))
+
+        free_surface = ImplicitFreeSurface(gravitational_acceleration=0.1)
+        coriolis = HydrostaticSphericalCoriolis(scheme = VectorInvariantEnstrophyConserving())
+
+        surface_wind_stress_parameters = (τ₀ = 1e-4,
+                                          Lφ = grid.Ly,
+                                          φ₀ = 15)
+        surface_wind_stress(λ, φ, t, p) = p.τ₀ * cos(2π * (φ - p.φ₀) / p.Lφ)
+
+        surface_wind_stress_bc = FluxBoundaryCondition(surface_wind_stress,
+                                               parameters = surface_wind_stress_parameters)
+
+        μ = 1 / 60days
+
+        @inline u_bottom_drag(i, j, grid, clock, fields, μ) = @inbounds - μ * fields.u[i, j, 1]
+        @inline v_bottom_drag(i, j, grid, clock, fields, μ) = @inbounds - μ * fields.v[i, j, 1]
+
+        u_bottom_drag_bc = FluxBoundaryCondition(u_bottom_drag,
+                                         discrete_form = true,
+                                         parameters = μ)
+
+        v_bottom_drag_bc = FluxBoundaryCondition(v_bottom_drag,
+                                         discrete_form = true,
+                                         parameters = μ)
+
+        u_bcs = FieldBoundaryConditions(top = surface_wind_stress_bc,
+                                bottom = u_bottom_drag_bc)
+
+        v_bcs = FieldBoundaryConditions(bottom = v_bottom_drag_bc)
+
+        νh₀ = 5e3 * (60 / grid.Nx)^2
+        νh(λ, φ, z, t) = νh₀ * cos(π * φ / 180)
+        variable_horizontal_diffusivity = HorizontallyCurvilinearAnisotropicDiffusivity(νh=νh)
```

Comment:
what do these lines do? seems that are not used further down...

Reply:
oops - just removed and also cherry picked original commits where I could